### PR TITLE
Some fixes in the Signals handling

### PR DESCRIPTION
**Pull request by thiblahute**
Namely:
  - Make all the Signal related APIs usable as Object methods
  - Fix signal_handler_(un)block arguments

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Wo nice!

**Comment by sdroege** [comment]
:+1: as discussed on irc earlier :) thanks!

File: src/object.rs [hunk 1]
```diff
@@ -566,6 +566,11 @@ pub trait ObjectExt: IsA<Object> {
     fn has_property<'a, N: Into<&'a str>>(&self, property_name: N, type_: Option<Type>) -> Result<(), BoolError>;
     fn get_property_type<'a, N: Into<&'a str>>(&self, property_name: N) -> Option<Type>;
 
+    fn block_signal(&self, handler_id: &SignalHandlerId);
+    fn unblock_signal(&self, handler_id: &SignalHandlerId);
+    fn stop_signal_emission(&self, signal_id: u32, detail: glib_ffi::GQuark);
```
Comment (sdroege) [inline src/object.rs]: No ffi types in the API here. We need GQuark bindings, or pass &str here and convert internally

Comment (thiblahute) [reply src/object.rs]: Should I just expose `stop_signal_emission_by_name` then? Renamed to `stop_signal_emission` maybe?

Comment (sdroege) [reply src/object.rs]: Yeah, let's do the quark stuff once we have bindings for that. That's also how I did it for the signal emitter and connector function

Comment (sdroege) [reply src/object.rs]: I mean, also rename for now. We can break API later once quarks are there

File: src/object.rs [hunk 2]
```diff
@@ -566,6 +566,11 @@ pub trait ObjectExt: IsA<Object> {
...
+    fn stop_signal_emission_by_name(&self, signal_name: &str);
+
     fn connect<'a, N, F>(&self, signal_name: N, after: bool, callback: F) -> Result<SignalHandlerId, BoolError>
```
Comment (sdroege) [inline src/object.rs]: Should probably also add a `disconnect` here for completeness

File: src/signal.rs [hunk 3]
```diff
@@ -57,13 +57,13 @@ pub unsafe fn connect(receiver: *mut gobject_ffi::GObject, signal_name: &str, tr
     from_glib(handle)
 }
 
-pub fn signal_handler_block<T: IsA<Object>>(instance: &T, handler_id: SignalHandlerId) {
+pub fn signal_handler_block<T: IsA<Object>>(instance: &T, handler_id: &SignalHandlerId) {
```
Comment (EPashkin) [inline src/signal.rs]: Why `&` added?

Comment (EPashkin) [reply src/signal.rs]: Sorry, now understand, it because it non-copy.

File: src/object.rs (hunk 2 above)
Comment (thiblahute) [reply src/object.rs]: Done.

File: src/object.rs (hunk 1 above)
Comment (thiblahute) [reply src/object.rs]: Done.

**Comment by sdroege** [comment]
Good to be merged IMHO

**Comment by GuillaumeGomez** [comment]
For me too. What about you @EPashkin?

**Comment by EPashkin** [comment]
👍 too

**Comment by sdroege** [comment]
@GuillaumeGomez let's get it in then :)

**Pull request merged**